deltachat-ios/MessageKit/Views/Cells/FileMessageCell.swift

```diff
@@ -0,0 +1,138 @@
+import UIKit
+
+// A subclass of `MessageContentCell` used to display mixed media messages.
+open class FileMessageCell: MessageContentCell {
+
+    public static let insetBottom: CGFloat = 12
+    public static let insetHorizontalBig: CGFloat = 23
+    public static let insetHorizontalSmall: CGFloat = 12
+
+    // MARK: - Properties
+    var fileViewLeadingPadding: CGFloat = 0 {
+        didSet {
+            fileViewLeadingAlignment.constant = fileViewLeadingPadding
+        }
+    }
+
+    private lazy var fileViewLeadingAlignment: NSLayoutConstraint = {
+        return fileView.constraintAlignLeadingTo(messageContainerView, paddingLeading: 0)
+    }()
+
+
+    /// The `MessageCellDelegate` for the cell.
+    open override weak var delegate: MessageCellDelegate? {
+        didSet {
+            messageLabel.delegate = delegate
+        }
+    }
+
+    /// The label used to display the message's text.
+    open var messageLabel = MessageLabel()
+
+    lazy var fileView: FileView = {
+        let fileView = FileView()
+        fileView.translatesAutoresizingMaskIntoConstraints = false
+        return fileView
+    }()
+
+    // MARK: - Methods
+
+    /// Responsible for setting up the constraints of the cell's subviews.
+    open func setupConstraints(for messageKind: MessageKind) {
+        messageContainerView.removeConstraints(messageContainerView.constraints)
+
+        let fileViewConstraints = [fileView.constraintHeightTo(FileView.defaultHeight),
+                                    fileViewLeadingAlignment,
+                                    fileView.constraintAlignTrailingTo(messageContainerView),
+                                    fileView.constraintAlignTopTo(messageContainerView),
```

Comment:
I think we should add a bottomConstraint, too. 

Needed to have a self sizing cell.

Reply:
we have a fixed height here, it already defines the bottom implicitly, right?!

Reply:
this is only needed for self sizing cells - so can be ignored.